src/routes/transactions/transactions.service.ts

```diff
@@ -38,8 +48,40 @@ export class TransactionsService {
     private readonly queuedItemsMapper: QueuedItemsMapper,
     private readonly transactionsHistoryMapper: TransactionsHistoryMapper,
     private readonly transactionPreviewMapper: TransactionPreviewMapper,
+    private readonly moduleTransactionDetailsMapper: ModuleTransactionDetailsMapper,
+    private readonly multisigTransactionDetailsMapper: MultisigTransactionDetailsMapper,
+    private readonly transferDetailsMapper: TransferDetailsMapper,
   ) {}
 
+  async getById(chainId: string, txId: string): Promise<TransactionDetails> {
+    const [txType, safeAddress, id] = txId.split(TRANSACTION_ID_SEPARATOR);
+
+    if (txType === MODULE_TRANSACTION_PREFIX) {
+      const tx = await this.safeRepository.getModuleTransaction(chainId, id);
+      const safe = await this.safeRepository.getSafe(chainId, safeAddress);
+      return this.moduleTransactionDetailsMapper.mapDetails(chainId, tx, safe);
+    }
+    if (txType === TRANSFER_PREFIX) {
+      const transfer = await this.safeRepository.getTransfer(chainId, id);
+      const safe = await this.safeRepository.getSafe(chainId, safeAddress);
+      return this.transferDetailsMapper.mapDetails(chainId, transfer, safe);
+    }
+    if (txType === MULTISIG_TRANSACTION_PREFIX) {
+      const tx = await this.safeRepository.getMultiSigTransaction(chainId, id);
+      const safe = await this.safeRepository.getSafe(chainId, safeAddress);
```

Comment:
Nit: as above. I'd try to wrap each of these in a `Promise.all`.

Reply:
Done in [e919dcc](https://github.com/safe-global/safe-client-gateway-nest/pull/365/commits/e919dcc15402b90a898b9bcc55f074b4303e365f)